tests/Unit/DataStructures/VectorImplTestHelper.hpp

```diff
@@ -2,17 +2,78 @@
 // See LICENSE.txt for details.
 #pragma once
 
+#include <array>
 #include <cstddef>
 #include <memory>  // IWYU pragma: keep
 #include <tuple>
 #include <utility>
 
+#include "DataStructures/VectorImpl.hpp"
 #include "Utilities/Gsl.hpp"
 #include "Utilities/Requires.hpp"
 #include "Utilities/Tuple.hpp"
+#include "tests/Unit/TestingFramework.hpp"
 
 namespace TestHelpers {
 namespace VectorImpl {
+
+namespace detail {
+// `check_vectors` checks equality of vectors and related data types for tests.
+//
+// Comparisons between any pairs of the following are supported:
+// vector types
+// arithmetic types
+// arrays of vectors
+// arrays of arithmetic types
+
+// between two vectors
+template <typename T1, typename T2, typename VT1, typename VT2>
+inline void check_vectors(const ::VectorImpl<T1, VT1>& t1,
+                          const ::VectorImpl<T2, VT2>& t2) noexcept {
+  CHECK_ITERABLE_APPROX(VT1{t1}, VT2{t2});
+}
+// between two arithmetic types
+template <typename T1, typename T2,
+          Requires<cpp17::is_arithmetic_v<T1> and cpp17::is_arithmetic_v<T2>> =
+              nullptr>
+inline void check_vectors(const T1& t1, const T2& t2) noexcept {
+  CHECK(approx(t1) == t2);
+}
+// between an arithmetic type and a vector
+template <typename T1, typename T2, typename VT2,
+          Requires<cpp17::is_arithmetic_v<T1>> = nullptr>
+void check_vectors(const T1& t1, const ::VectorImpl<T2, VT2>& t2) noexcept {
+  check_vectors(VT2{t2.size(), t1}, t2);
+}
+// between a vector and an arithmetic type
+template <typename T1, typename VT1, typename T2,
+          Requires<cpp17::is_arithmetic_v<T2>> = nullptr>
+void check_vectors(const ::VectorImpl<T1, VT1>& t1, const T2& t2) noexcept {
+  check_vectors(t2, t1);
+}
+// between two arrays
+// between two arrays
```

Comment:
I guess one of these comments probably suffices :)

Reply:
D'oh >.<